Ecobee logging cleanup

bjpetit opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


## Proposed change
Remove logging from ecobee property routines

Consolidate unknown thermostat model error logging into the
climate initializer

Modify each ecobee entity to set the model to `None`
in the device_info property, rather than failing the call. 


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

bjpetit committed:
Remove logging from ecobee property routines

Consolidate unknown thermostat model error logging into the
climate initializer

Modify each ecobee entity to set the model to "Unknown Thermostat"
in the device_info property, rather than failing the call

bjpetit commented:
Personally I wonder if it would be better to have the error message direct folks to create a new issue, rather continue to resurrect a closed issue. It feels like creating a new ticket makes more sense at this point.

MartinHjelmare commented on homeassistant/components/ecobee/binary_sensor.py:
```diff
@@ -67,14 +67,7 @@ def device_info(self):
-                        thermostat["name"],
-                        thermostat["modelNumber"],
-                    )
+                    model = "Unknown Thermostat"
```
If we don't know the model we set it to `None` or don't include it in the device info at all.

bjpetit replied:
Updated to set the model to `None` if unknown.

MartinHjelmare commented on homeassistant/components/ecobee/climate.py:
```diff
@@ -300,6 +300,17 @@ def __init__(self, data, thermostat_index):
 
+        # Check that this is a known model, log message if not
+        if not self.thermostat["modelNumber"] in ECOBEE_MODEL_TO_NAME:
+            _LOGGER.error(
```
We don't want side effect like logging in init methods.

bjpetit replied:
I've moved this out of the init method. It is now in a Throttled method called by `async_update`. The current setting logs errors once an hour to avoid too much log polution.

MartinHjelmare commented:
Yes, we prefer new issues over catch all issues.

bjpetit committed:
Update unknown ecobee model handling

Update error message with github new issue link
rather than pointing to closed issue

Update device_info routines to set model to None
if not in model list

bjpetit committed:
Move unknown ecobee model logging out of init
method into throttled method called by async_update

bjpetit commented:
> Yes, we prefer new issues over catch all issues.

Thanks, I've updated the text to reflect this.

MartinHjelmare commented on homeassistant/components/ecobee/climate.py:
```diff
@@ -330,6 +335,23 @@ async def async_update(self):
+        await self.throttled_validate_data()
+
+    @Throttle(VALIDATE_DATA_INTERVAL)
+    async def throttled_validate_data(self):
```
This doesn't need to be a coroutine function since we don't await inside.

bjpetit replied:
I've moved the check into `async_setup_entry`. So `throttled_validate_data` goes away.

MartinHjelmare commented on homeassistant/components/ecobee/climate.py:
```diff
@@ -330,6 +335,23 @@ async def async_update(self):
+    async def throttled_validate_data(self):
+        """Validate data from ecobee periodically and log errors."""
+        # Check that this is a known model, log message if not
+        if not self.thermostat["modelNumber"] in ECOBEE_MODEL_TO_NAME:
```
I suggest we move this check to just before we create the entity. Make a for loop in `async_setup_entry`.
```py
entities = []

for index in range(len(data.ecobee.thermostats)):
    thermostat = data.ecobee.get_thermostat(index)
    if not thermostat["modelNumber"] in ECOBEE_MODEL_TO_NAME:
        ...
    entities.append(Thermostat(data, index, thermostat))
```

bjpetit replied:
Thanks. The check has been moved into `async_setup_entry`. I also needed to clean up the references to `device` in some of the subsequent service definitions.

bjpetit committed:
Moved unknown thermostat check into async_setup_entry

MartinHjelmare commented on homeassistant/components/ecobee/climate.py:
```diff
@@ -176,18 +176,31 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
+                thermostat["name"],
+                thermostat["modelNumber"],
+            )
+        entities.append(Thermostat(data, index))
```
If we pass the thermostat to the entity we save a call in the entity init.
```suggestion
        entities.append(Thermostat(data, index, thermostat))
```

bjpetit committed:
Pass thermostat into ecobee Thermostat initializer

MartinHjelmare reviewed (approved):
Looks good!
